plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2924,54 +2924,79 @@ void read_write::send_transaction(const read_write::send_transaction_params& par
+}
 
-         if (std::holds_alternative<fc::exception_ptr>(result)) {
-            auto& eptr = std::get<fc::exception_ptr>(result);
-            fc_add_tag(trx_span, "error", eptr->to_string());
-            next(eptr);
-         } else {
-            auto& trx_trace_ptr = std::get<transaction_trace_ptr>(result);
+void read_write::send_transaction(const read_write::send_transaction_params_v2& params, next_function<read_write::send_transaction_results> next) {
 
-            fc_add_tag(trx_span, "block_num", trx_trace_ptr->block_num);
-            fc_add_tag(trx_span, "block_time", trx_trace_ptr->block_time.to_time_point());
-            fc_add_tag(trx_span, "elapsed", trx_trace_ptr->elapsed.count());
-            if( trx_trace_ptr->receipt ) {
-               fc_add_tag(trx_span, "status", std::string(trx_trace_ptr->receipt->status));
-            }
-            if( trx_trace_ptr->except ) {
-               fc_add_tag(trx_span, "error", trx_trace_ptr->except->to_string());
-            }
+   try {
+      packed_transaction_v0 input_trx_v0;
+      auto resolver = make_resolver(db, abi_serializer::create_yield_function( abi_serializer_max_time ));
+      packed_transaction_ptr input_trx;
+      try {
+         abi_serializer::from_variant(params.transaction, input_trx_v0, std::move( resolver ), abi_serializer::create_yield_function( abi_serializer_max_time ));
+         input_trx = std::make_shared<packed_transaction>( std::move( input_trx_v0 ), true );
+      } EOS_RETHROW_EXCEPTIONS(chain::packed_transaction_type_exception, "Invalid packed transaction")
 
-            try {
-               fc::variant output;
-               try {
-                  output = db.to_variant_with_abi( *trx_trace_ptr, abi_serializer::create_yield_function( abi_serializer_max_time ) );
-               } catch( chain::abi_exception& ) {
-                  output = *trx_trace_ptr;
-               }
+      send_transaction(input_trx, "/v2/chain/send_transaction", params.return_failure_traces, next);
 
-               const chain::transaction_id_type& id = trx_trace_ptr->id;
-               next(read_write::send_transaction_results{id, output});
-            } CATCH_AND_CALL(next);
-         }
-      });
    } catch ( boost::interprocess::bad_alloc& ) {
       chain_plugin::handle_db_exhaustion();
    } catch ( const std::bad_alloc& ) {
       chain_plugin::handle_bad_alloc();
    } CATCH_AND_CALL(next);
 }
 
+void read_write::send_transaction(packed_transaction_ptr input_trx, const std::string method, bool return_failure_traces, chain::plugin_interface::next_function<send_transaction_results> next) {
+   auto trx_trace = fc_create_trace_with_id("Transaction", input_trx->id());
+   auto trx_span = fc_create_span(trx_trace, "HTTP Received");
+   fc_add_tag(trx_span, "trx_id", input_trx->id());
+   fc_add_tag(trx_span, "method", method);
+
+   app().get_method<incoming::methods::transaction_async>()(input_trx, true, false, static_cast<const bool>(return_failure_traces),
```

Comment:
Why `static_cast<const bool>(return_failure_traces)`, the cast does not seem needed.

Reply:
You'd think so, but it won't build without it.  I don't understand why either, but the way the `get_method<>` template works, the cast is required.

Reply:
Ah, template method, missed that. That make sense.